### PR TITLE
feat(table): enhance filter capabilities

### PR DESCRIPTION
**Pull request by georgettica**
- fix(spinner): make code compile
- feat(table): make filtering a third party

filter now uses JMESPath that seems very powerfull

### CODE REVIEW

**Comment by georgettica** [comment]
I forgot to fix the tests.. doing this now

**Comment by georgettica** [comment]
so I need assistance now..

I know the code should be working but it doesn't.
I was not able to fix this test easily and I don't want to disable that test without confirmation (as it seems to look ok)

File: test/styled/table.test.ts [hunk 1]
```diff
@@ -283,15 +283,17 @@ describe('styled/table', () => {
...
-    fancy
-    .stdout()
-    .end('ignores header case', output => {
-      cli.table(apps, columns, {columns: 'iD,Name', filter: 'nAMe=supertable-test', sort: '-ID'})
-      expect(output.stdout).to.equal(` ID  Name${ws.padEnd(14)}
- ─── ─────────────────${ws}
- 321 supertable-test-2${ws}
- 123 supertable-test-1${ws}\n`)
-    })
+// Why is this a use case we need to check?
```
Comment (RodEsp) [inline test/styled/table.test.ts:286]: So that someone doesn't accidentally make `--filter` case sensitive.

Comment (georgettica) [reply test/styled/table.test.ts:286]: So now filter is case sensitive, how much of a problem that is?

**Comment by georgettica** [comment]
@RodEsp closing as I gave an alternative solution  https://github.com/oclif/cli-ux/pull/447

**Pull request closed without merging**